In [45]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import accuracy_score,confusion_matrix
from sklearn.metrics import classification_report
from sklearn.metrics import mean_squared_error,mean_absolute_error

In [2]:
train= pd.read_csv(r"C:\Users\dtdee\OneDrive\Desktop\Letsupgrade_Python\Machine_Learning\Linear_Regression\1678600978303_train.csv")
test=pd.read_csv(r"C:\Users\dtdee\OneDrive\Desktop\Letsupgrade_Python\Machine_Learning\Linear_Regression\1678600974300_test.csv")

In [10]:
train.head()

x          y
0  24.0  21.549452
1  50.0  47.464463
2  15.0  17.218656
3  38.0  36.586398
4  87.0  87.288984

In [4]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   x       700 non-null    float64
 1   y       699 non-null    float64
dtypes: float64(2)
memory usage: 11.1 KB


In [5]:
train.isnull().sum()

x    0
y    1
dtype: int64

In [11]:
train.dropna(inplace=True)

In [12]:
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 699 entries, 0 to 699
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   x       699 non-null    float64
 1   y       699 non-null    float64
dtypes: float64(2)
memory usage: 16.4 KB


In [31]:
X_train=train.iloc[:,:-1].values
y_train=train.iloc[:,1:].values
X_test=test.iloc[:,:-1].values
y_test=test.iloc[:,1:].values

In [34]:
reg=LinearRegression()

In [37]:
reg.fit(X_train,y_train)

LinearRegression()

In [38]:
y_pred= reg.predict(X_test)

In [39]:
print(y_pred[0:10])
print()
print()
print(y_test[0:10])

[[76.94327594]
 [20.90651855]
 [21.90717494]
 [19.90586217]
 [35.91636428]
 [14.90258026]
 [61.93343021]
 [94.95509081]
 [19.90586217]
 [ 4.89601644]]


[[79.77515201]
 [23.17727887]
 [25.60926156]
 [17.85738813]
 [41.84986439]
 [ 9.80523488]
 [58.87465933]
 [97.61793701]
 [18.39512747]
 [ 8.74674765]]


### MODEL EVALUATION

In [48]:
mean_absolute_error(y_test,y_pred)

2.415771850041258

In [49]:
mean_squared_error(y_test,y_pred)

9.432922192039305

In [51]:
np.sqrt(mean_squared_error(y_test,y_pred))

3.071306268029827

### Conclusion::

All of our errors are low - and we're missing the actual value by 3.07
at most (lower or higher), which is a pretty small range considering the data we have.

### Checking the model preformance and score

In [52]:
reg.score(X_train,y_train)

0.9907015319025518

In [53]:
reg.score(X_test,y_test)

0.9888014444327563

In [54]:
reg.score(y_pred,y_test)

0.9887517938449388